src/go/rdctl/pkg/shell/shell.go

```diff
@@ -0,0 +1,165 @@
+package shell
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"runtime"
+	"slices"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"golang.org/x/text/encoding/unicode"
+
+	"github.com/rancher-sandbox/rancher-desktop/src/go/rdctl/pkg/directories"
+	"github.com/rancher-sandbox/rancher-desktop/src/go/rdctl/pkg/lima"
+	"github.com/rancher-sandbox/rancher-desktop/src/go/rdctl/pkg/paths"
+	"github.com/rancher-sandbox/rancher-desktop/src/go/rdctl/pkg/utils"
+	"github.com/rancher-sandbox/rancher-desktop/src/go/rdctl/pkg/wsl"
+)
+
+// Spawn a command that, when run, will be executed in the VM with the given
+// arguments.
+func SpawnCommand(ctx context.Context, args ...string) (*exec.Cmd, error) {
+	commandName, err := directories.GetLimactlPath()
+	if err != nil {
+		return nil, err
+	}
+
+	if runtime.GOOS == "windows" {
+		distroNames := []string{wsl.DistributionName}
+		found := false
+
+		if _, err = os.Stat(commandName); err == nil {
+			// If limactl is available, try the lima distribution first.
+			distroNames = append([]string{lima.InstanceFullName}, distroNames...)
+		}
+
+		for _, distroName := range distroNames {
+			err = assertWSLIsRunning(ctx, distroName)
+			if err == nil {
+				commandName = "wsl"
+				args = append([]string{
+					"--distribution", distroName,
+					"--exec", "/usr/local/bin/wsl-exec",
+				}, args...)
+				found = true
+				break
+			}
+		}
+
+		if !found {
+			// We did not find a running distribution that we can use.
+			// No further output wanted, so just exit with the desired status.
+			fmt.Fprintf(os.Stderr, "%s", err)
+			os.Exit(1)
+		}
+	} else {
+		p, err := paths.GetPaths()
+		if err != nil {
+			return nil, err
+		}
+		if err := directories.SetupLimaHome(p.AppHome); err != nil {
+			return nil, err
+		}
+		if err := setupPathEnvVar(p); err != nil {
+			return nil, err
+		}
+		if !checkLimaIsRunning(ctx, commandName) {
+			// No further output wanted, so just exit with the desired status.
+			os.Exit(1)
+		}
+		args = append([]string{"shell", lima.InstanceName}, args...)
+	}
+	return exec.CommandContext(ctx, commandName, args...), nil
+}
+
+// Set up the PATH environment variable for limactl.
+func setupPathEnvVar(p *paths.Paths) error {
+	if runtime.GOOS != "windows" {
+		// This is only needed on Windows.
+		return nil
+	}
+	msysDir := filepath.Join(utils.GetParentDir(p.Resources, 2), "msys")
+	pathList := filepath.SplitList(os.Getenv("PATH"))
+	if slices.Contains(pathList, msysDir) {
+		return nil
+	}
+	pathList = append([]string{msysDir}, pathList...)
+	return os.Setenv("PATH", strings.Join(pathList, string(os.PathListSeparator)))
+}
+
+const restartDirective = "Either run 'rdctl start' or start the Rancher Desktop application first"
```

Comment:
I'm not sure which style Rancher uses...
```suggestion
const restartDirective = "Either run `rdctl start` or start the Rancher Desktop application first"
```

Reply:
Actually, going through the rest of `rdctl` we seem to use ``` ' ``` over ``` ` ``` so I'll stick with the apostrophe…

Also, I did not in fact add any backtick-style quoting in the PR, oops.